swupd/manifest.go

```diff
@@ -882,35 +885,39 @@ func writeIndexManifest(c *config, ui *UpdateInfo, bundles []*Manifest) (*Manife
 	oldM.sortFilesName()
 	// linkPeersAndChange will update file versions correctly
 	_, _, _ = idxMan.linkPeersAndChange(oldM, ui.minVersion)
-	// now add any new files to the full manifest
-	for _, idxF := range idxMan.Files {
-		i := sort.Search(len(newFull.Files), func(i int) bool {
-			return newFull.Files[i].Name >= idxF.Name
-		})
-
-		if i < len(newFull.Files) && newFull.Files[i].Name == idxF.Name {
-			// overwrite existing
-			newFull.Files[i] = idxF
-			continue
-		}
 
-		// add to full manifest
-		newFull.Files = append(newFull.Files, idxF)
-	}
+	idxMan.Header.FileCount = uint32(len(idxMan.Files))
 
-	// done processing, sort by version before writing
-	idxMan.sortFilesVersionName()
-	// check that there is actually a change to this manifest before writing
-	if idxMan.Files[len(idxMan.Files)-1].Version < idxMan.Header.Version {
-		// return the old version of the manifest since there was no change
-		// in this version
-		return oldM, nil
-	}
+	if idxMan.Header.FileCount > 0 {
```

Comment:
Since you are at it, this will be cleaner if you just return if FileCount <= 0 instead of moving all the code inside a block.